internal/system/agent/services.go

```diff
@@ -51,114 +53,208 @@ func InvokeOperation(action string, services []string) bool {
+					} else {
+						c.Configuration[service] = ProcessResponse(responseJSON)
+					}
+					return c, nil
+				}
+			}
 		} else {
-			configuration[service] = ProcessResponse(responseJSON)
+			// Service is known to SMA, so no need to ask the Registry for a ServiceEndpoint associated with `service`
+			// Simply use one of the ready-made list of clients.
+			LoggingClient.Info(fmt.Sprintf("service %s is known to SMA as being in the ready-made list of clients", service))
+
+			c.Configuration[service] = ""
+
+			responseJSON, err := generalClients[service].FetchConfiguration(ctx)
+			if err != nil {
+				c.Configuration[service] = fmt.Sprintf("%s get configuration error: %s", service, err.Error())
+				LoggingClient.Error("error retrieving configuration", "service name", service, "error message", err.Error())
+			} else {
+				c.Configuration[service] = ProcessResponse(responseJSON)
+			}
 		}
 	}
-	return configuration, nil
+	return c, nil
 }
 
-func getMetrics(services []string, ctx context.Context) (map[string]interface{}, error) {
+func getMetrics(services []string, ctx context.Context) (MetricsRespMap, error) {
 
-	metrics := make(map[string]interface{})
+	m := MetricsRespMap{}
+	m.Metrics = map[string]interface{}{}
 
-	// Loop through requested actions, along with respectively-supplied parameters.
+	// Loop through requested actions, along with (any) respectively-supplied parameters.
 	for _, service := range services {
 
+		// Check whether SMA does _not_ know of ServiceKey ("service") as being one for one of its ready-made list of clients.
 		if !isKnownServiceKey(service) {
-			logs.LoggingClient.Warn("unknown service found getting metrics", "service name", service)
-		}
+			LoggingClient.Info(fmt.Sprintf("service %s not known to SMA as being in the ready-made list of clients", service))
+
+			// Service unknown to SMA, so ask the Registry whether `service` is available.
+			serviceError := registryClient.IsServiceAvailable(service)
+			if serviceError != nil {
+				msg := fmt.Sprintf("Registry responded with %s service not available, with error code: %s", service, serviceError.Error())
+				LoggingClient.Error(msg)
+				m.Metrics[service] = fmt.Sprintf(msg)
+			} else {
+				LoggingClient.Info(fmt.Sprintf("Registry responded with %s service available", service))
 
-		responseJSON, err := clients[service].FetchMetrics(ctx)
-		if err != nil {
-			metrics[service] = fmt.Sprintf("%s get metrics error: %s", service, err.Error())
-			logs.LoggingClient.Error("error retrieving metrics", "service name", service, "errMsg", err.Error())
+				// Since service is unknown to SMA, ask the Registry for a ServiceEndpoint associated with `service`
+				e, err := registryClient.GetServiceEndpoint(service)
+				if err != nil {
+					msg := fmt.Sprintf("on attempting to get ServiceEndpoint for service %s, got error: %v", service, err)
+					LoggingClient.Error(msg)
+					return m, err
```

Comment:
Same comment w/r/t returning if caller is asking for metrics using multiple keys